service_crategen/src/commands/generate/codegen/xml_payload_parser.rs

```diff
@@ -245,7 +245,12 @@ fn generate_deserializer_body(name: &str, shape: &Shape, service: &Service) -> S
         ShapeType::List => generate_list_deserializer(shape, service),
         ShapeType::Map => generate_map_deserializer(shape),
         ShapeType::Structure => generate_struct_deserializer(name, service, shape),
-        _ => generate_primitive_deserializer(shape),
+
+        // For mysterious historical reasons, all policies returned by the IAM APIs are URI-encoded, and
```

Comment:
Could we drop the `For mysterious historical reasons` part of this comment? Since we don't know why IAM is the way it is, we can stick with the facts it's a bit special. 😄